app/src/main/java/com/example/android_level_3/FragmentAddContact.kt

```diff
@@ -0,0 +1,219 @@
+            }
+        }
+
+        viewModel.listWithAllContacts.observe(viewLifecycleOwner) { list ->
+            // при обновлении списка контактов, обновляем контакт-лист (для отображения добавленых)
+            list.forEach {
+                viewModel.userContactsIdList.add(it.id)
+            }
+
+            if(viewModel.isActiveSearchAddContact.value == false) {
+                createAdapter(multiSelectState = null, usersInContactList = viewModel.userContactsIdList.toList())
+            }
+        }
+
+        viewModel.addToContactListResult.observe(viewLifecycleOwner) { serverResponse ->
+            if (serverResponse == null) {
+                connectionErrorSnackbar = createConnectionErrorSnackBar(Const.REQUEST_ADD_CONTACT)
+                connectionErrorSnackbar?.show()
+            } else if (serverResponse.isSuccessful) {
+                viewModel.listWithAllContacts.value = serverResponse.body()?.data?.contacts
+            } else {                                                                                // если неудачная авторизации (перезапускаем добавление контакта)
+                connectionErrorSnackbar = createConnectionErrorSnackBar(Const.REQUEST_ADD_CONTACT)
+                connectionErrorSnackbar?.show()
+            }
+        }
+
+        viewModel.getAllUsersResult.observe(viewLifecycleOwner) { serverResponse ->
+            if (serverResponse == null) {
+                connectionErrorSnackbar = createConnectionErrorSnackBar(Const.REQUEST_GET_ALL_USERS)
+                connectionErrorSnackbar?.show()
+            } else if (serverResponse.isSuccessful) {
+                viewModel.listOfAllUsers.value = serverResponse.body()?.data?.users
+            } else {                                                                                // если неудачная авторизации (перезапускаем)
+                connectionErrorSnackbar = createConnectionErrorSnackBar(Const.REQUEST_GET_ALL_USERS)
+                connectionErrorSnackbar?.show()
+            }
+        }
+
+        isVisibleProgressBar.observe(viewLifecycleOwner) { visibility ->
+            if (visibility) binding.progressBar.visibility = View.VISIBLE
+            else binding.progressBar.visibility = View.GONE
+        }
+
+        // показываем/прячем панель для поиска (при изменении состояния)
+        viewModel.isActiveSearchAddContact.observe(viewLifecycleOwner) { visible ->
+            if (visible) {
+                with(binding){
+                    toolbarAddContact.containerTextAddContact.visibility = View.GONE
+                    toolbarAddContact.containerSearchAddContact.visibility = View.VISIBLE
+                }
+            } else {
+                with(binding){
+                    toolbarAddContact.containerTextAddContact.visibility = View.VISIBLE
+                    toolbarAddContact.containerSearchAddContact.visibility = View.GONE
+                    toolbarAddContact.edSearchAddContact.setText("")
+                }
```

Comment:
you can use "with" function outside the conditional statement and therefor simplify the code